15/root/usr/share/container-scripts/postgresql/README.md

```diff
@@ -0,0 +1,331 @@
+post-migration time).
+
+Please keep in mind that the container image provides help for users'
+convenience, but fully automatic migration is not guaranteed.  Thus, before you
+start proceeding with the database migration, get prepared to perform manual
+steps in order to get all your data migrated.
+
+Note that you might not use variables like `POSTGRESQL_USER` in migration
+scenario, all the data (including info about databases, roles or passwords are
+copied from old cluster).  Ensure that you use the same
+`OPTIONAL_CONFIGURATION_VARIABLES` as you used for initialization of the old
+PostgreSQL container.  If some non-default configuration is done on remote
+cluster, you might need to copy the configuration files manually, too.
+
+Security warning:  Note that the IP communication between old and new PostgreSQL
+clusters is not encrypted by default, it is up to user to configure SSL on
+remote cluster or ensure security via different means.
+
+PostgreSQL auto-tuning
+----------------------
+
+When the PostgreSQL image is run with the `--memory` parameter set and if there
+are no values provided for `POSTGRESQL_SHARED_BUFFERS` and
+`POSTGRESQL_EFFECTIVE_CACHE_SIZE` those values are automatically calculated
+based on the value provided in the `--memory` parameter.
+
+The values are calculated based on the
+[upstream](https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server)
+formulas. For the `shared_buffers` we use 1/4 of given memory and for the
+`effective_cache_size` we set the value to 1/2 of the given memory.
+
+PostgreSQL admin account
+------------------------
+The admin account `postgres` has no password set by default, only allowing local
+connections.  You can set it by setting the `POSTGRESQL_ADMIN_PASSWORD` environment
+variable when initializing your container. This will allow you to login to the
+`postgres` account remotely. Local connections will still not require a password.
+
+
+Changing passwords
+------------------
+
+Since passwords are part of the image configuration, the only supported method
+to change passwords for the database user (`POSTGRESQL_USER`) and `postgres`
+admin user is by changing the environment variables `POSTGRESQL_PASSWORD` and
+`POSTGRESQL_ADMIN_PASSWORD`, respectively.
+
+Changing database passwords through SQL statements or any way other than through
+the environment variables aforementioned will cause a mismatch between the
+values stored in the variables and the actual passwords. Whenever a database
+container starts it will reset the passwords to the values stored in the
+environment variables.
+
+
+Upgrading database (by switching to newer PostgreSQL image version)
+-------------------------------------------------------------------
+
+** Warning! Please, before you decide to do the data directory upgrade, always
+ensure that you've carefully backed up all your data and that you are OK with
+potential manual rollback! **
+
+This image supports automatic upgrade of data directory created by
```

Comment:
Is this true? Which script does it?

Reply:
Good point. It's the common.sh, but the in-place upgrade might only work on SCL, so we should remove this upgrade section in the README.md on systems other then SCL (RHEL-7). For RHEL-8+, the only way to move to a higher version is to use the "migration" from a remote server (a different container).

Reply:
I'm adding `+{% if spec.prod == 'rhel7' or spec.prod == 'centos7' %}` condition, so README.md will not include this for RHEL-8+.